lib/board.rb

```diff
@@ -0,0 +1,49 @@
+class Board
+
+    attr_accessor :cells, :turn_count
+
+    def initialize
+        self.reset!
+    end
+
+    def reset!
+        self.cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    end
+
+    def display
+        puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+        puts "-----------"
+        puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+        puts "-----------"
+        puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+    end
+
+    def position(input)
+        self.cells[input.to_i - 1]
+    end
+
+    def full?
+        self.cells.all? {|cell| cell == "X" || cell == "O"}
+    end
+
+    def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```